diff --git a/node_modules/matrix-js-sdk/src/http-api/fetch.ts b/node_modules/matrix-js-sdk/src/http-api/fetch.ts
index 4285d92..c644ae2 100644
--- a/node_modules/matrix-js-sdk/src/http-api/fetch.ts
+++ b/node_modules/matrix-js-sdk/src/http-api/fetch.ts
@@ -210,6 +210,12 @@ export class FetchHttpApi<O extends IHttpOpts> {
             } else if (error.errcode == "M_CONSENT_NOT_GIVEN") {
                 this.eventEmitter.emit(HttpApiEvent.NoConsent, error.message, error.data.consent_uri);
             }
+            //:tchap: this error is sent by the synapse module (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity]
+            // when user account has expired
+            else if (error.errcode == 'ORG_MATRIX_EXPIRED_ACCOUNT') {
+                this.eventEmitter.emit(HttpApiEvent.ORG_MATRIX_EXPIRED_ACCOUNT, error);
+            }
+            //:tchap end
 
             throw error;
         }
diff --git a/node_modules/matrix-js-sdk/src/http-api/interface.ts b/node_modules/matrix-js-sdk/src/http-api/interface.ts
index bfa604c..c235575 100644
--- a/node_modules/matrix-js-sdk/src/http-api/interface.ts
+++ b/node_modules/matrix-js-sdk/src/http-api/interface.ts
@@ -104,6 +104,10 @@ export interface IContentUri {
 export enum HttpApiEvent {
     SessionLoggedOut = "Session.logged_out",
     NoConsent = "no_consent",
+    //:tchap: this error is sent by the synapse module (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity] 
+    //when user account has expired
+    ORG_MATRIX_EXPIRED_ACCOUNT = "ORG_MATRIX_EXPIRED_ACCOUNT",
+    //:tchap end
 }
 
 export type HttpApiEventHandlerMap = {
@@ -131,6 +135,9 @@ export type HttpApiEventHandlerMap = {
      * ```
      */
     [HttpApiEvent.NoConsent]: (message: string, consentUri: string) => void;
+    //:tchap: this error is sent by the synapse module (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity] 
+    //when user account has expired
+    [HttpApiEvent.ORG_MATRIX_EXPIRED_ACCOUNT]: (err: MatrixError) => void;
 };
 
 export interface UploadProgress {
diff --git a/node_modules/matrix-js-sdk/src/sync.ts b/node_modules/matrix-js-sdk/src/sync.ts
index 1419d1c..fd0c725 100644
--- a/node_modules/matrix-js-sdk/src/sync.ts
+++ b/node_modules/matrix-js-sdk/src/sync.ts
@@ -599,6 +599,14 @@ export class SyncApi {
             this.updateSyncState(SyncState.Error, { error });
             return true;
         }
+        //:tchap: this error is sent by (synapse-email-account-validity)[https://github.com/matrix-org/synapse-email-account-validity] 
+        // when user account has expired. client must be stopped and sync aborted when the user account is not valid anymore
+        if (error.errcode === "ORG_MATRIX_EXPIRED_ACCOUNT") {
+            // The account is expired
+            this.stop();
+            this.updateSyncState(SyncState.Error, { error });
+            return true;
+        }
         return false;
     }
 
